lib/fluent/plugin/out_logdna.rb

```diff
@@ -14,11 +14,16 @@ class LogDNAOutput < Fluent::BufferedOutput
     config_param :app, :string, default: nil
     config_param :file, :string, default: nil
     config_param :ingester_domain, :string, default: 'https://logs.logdna.com'
-    config_param :request_timeout_seconds, :integer, default: 30
+    config_param :request_timeout, :string, default: '30'
 
     def configure(conf)
       super
       @host = conf['hostname']
+
+      timeout_match = @request_timeout.match(/[0-9]*/)[0]
+      timeout = timeout_match == "" ? 30 : timeout_match.to_i
+      ms_match = @request_timeout.include? "ms"
+      @request_timeout = ms_match && timeout_match != "" ? timeout / 1000 : timeout
```

Comment:
It seems weird that we're differentiating millisecond vs second based on the presence of a number in the env var (`timeout_match != ""`). I think this was added to handle the case for an input of just `ms` which would default to `30 ms` which is a bit odd. Can't we just make prefix only inputs invalid and always default to `30s`?

I think it's mostly odd because we're handling weird cases even in the last line... I think it would be conceptually cleaner if we could format the inputs to be correct and then execute our logic on clean inputs

Reply:
Yeah, overly complicated way of covering for unit only config. Fixed with suggestion below